pkg/kubelet/pod/pod_manager_test.go

```diff
@@ -37,131 +40,185 @@ func newTestManager() (*basicManager, *podtest.FakeMirrorClient) {
+			needToRemovePod:  mirrorPod,
+			expectPods:       []*v1.Pod{normalPod, staticPod},
+			expectMirrorPods: []*v1.Pod{},
+		},
+		{
+			name:             "Remove static pod",
+			podList:          []*v1.Pod{mirrorPod, staticPod, normalPod},
+			needToRemovePod:  staticPod,
+			expectPods:       []*v1.Pod{normalPod},
+			expectMirrorPods: []*v1.Pod{mirrorPod},
+		},
+		{
+			name:             "Remove normal pod",
+			podList:          []*v1.Pod{mirrorPod, staticPod, normalPod},
+			needToRemovePod:  normalPod,
+			expectPods:       []*v1.Pod{staticPod},
+			expectMirrorPods: []*v1.Pod{mirrorPod},
 		},
-		staticPod,
-	}
-	updates := append(expectedPods, mirrorPod)
-	podManager, _ := newTestManager()
-	podManager.SetPods(updates)
-
-	podManager.RemovePod(staticPod)
-
-	actualPods := podManager.GetPods()
-	if len(actualPods) == len(expectedPods) {
-		t.Fatalf("Run RemovePod() error, expected %d pods, got %d pods; ", len(expectedPods)-1, len(actualPods))
-	}
-
-	_, _, orphanedMirrorPodNames := podManager.GetPodsAndMirrorPods()
-	expectedOrphanedMirrorPodNameNum := 1
-	if len(orphanedMirrorPodNames) != expectedOrphanedMirrorPodNameNum {
-		t.Fatalf("Run getOrphanedMirrorPodNames() error, expected %d orphaned mirror pods, got %d orphaned mirror pods; ", expectedOrphanedMirrorPodNameNum, len(orphanedMirrorPodNames))
 	}
 
-	expectedOrphanedMirrorPodName := mirrorPod.Name + "_" + mirrorPod.Namespace
-	if orphanedMirrorPodNames[0] != expectedOrphanedMirrorPodName {
-		t.Fatalf("Run getOrphanedMirrorPodNames() error, expected orphaned mirror pod name : %s, got orphaned mirror pod name %s; ", expectedOrphanedMirrorPodName, orphanedMirrorPodNames[0])
+	for _, test := range testCase {
+		t.Run(test.name, func(t *testing.T) {
+			podManager, _ := newTestManager()
+			podManager.SetPods(test.podList)
+			podManager.RemovePod(test.needToRemovePod)
+			actualPods1 := podManager.GetPods()
+			actualPods2, actualMirrorPods, _ := podManager.GetPodsAndMirrorPods()
+			// Sort the pods to make the test deterministic.
+			sort.Slice(actualPods1, func(i, j int) bool {
+				return actualPods1[i].Name < actualPods1[j].Name
+			})
+			sort.Slice(actualPods2, func(i, j int) bool {
+				return actualPods2[i].Name < actualPods2[j].Name
+			})
+			sort.Slice(test.expectPods, func(i, j int) bool {
+				return test.expectPods[i].Name < test.expectPods[j].Name
+			})
+
+			// Check if the actual pods and mirror pods match the expected ones.
+			if diff := cmp.Diff(actualPods1, actualPods2); diff != "" {
+				t.Errorf("Test case %s failed: actualPods1 and actualPods2 are not equal", test.name)
+			}
+			if diff := cmp.Diff(actualPods1, test.expectPods); diff != "" {
+				t.Errorf("Test case %s failed: actualPods1 and expectPods are not equal", test.name)
+			}
+			if diff := cmp.Diff(actualMirrorPods, test.expectMirrorPods); diff != "" {
+				t.Errorf("Test case %s failed: actualMirrorPods and expectMirrorPods are not equal", test.name)
+			}
```

Comment:
I'm not sure it's worth checking the actual pods returned by `podManager.GetPods()` and `podManager.GetPodsAndMirrorPods()` since they both return `podsMapToPods(pm.podByUID)` the same way. I think it’d be fine to verify it once.

Also, if you're using `cmp.Diff`, there’s an option to sort   slices, so I think you can write it more simply like this:

```suggestion
			actualPods := podManager.GetPods()
			_, actualMirrorPods, _ := podManager.GetPodsAndMirrorPods()

			sortOpt := cmpopts.SortSlices(func(a, b *v1.Pod) bool { return a.Name < b.Name })
			// Check if the actual pods and mirror pods match the expected ones.
			if diff := cmp.Diff(actualPods, test.expectPods, sortOpt); diff != "" {
				t.Errorf("actualPods and expectPods are not equal: %s", diff)
			}
			if diff := cmp.Diff(actualMirrorPods, test.expectMirrorPods, sortOpt); diff != "" {
				t.Errorf("actualMirrorPods and expectMirrorPods are not equal: %s", diff)
			}
```